xunit-autolabeler-v2/ast_parser/core/analyze.py

```diff
@@ -119,6 +133,57 @@ def _dedupe_source_methods(
     return list(source_methods_deduped)
 
 
+def _store_tests_on_methods(
+    source_methods: List[pdd.PolyglotDriftData],
+    test_to_method_key_map: Dict[Tuple[str, str], Tuple[str, str]]
+) -> None:
+    """Adds test data to snippet method objects
+
+    This method uses test keys and the mapping generated by per-language
+    parsers to match snippet methods with their tests. That test data
+    (file paths and method names) is then added to the methods themselves
+    by updating their "test_methods" property.
+
+    Args:
+        source_methods: a list of top-level methods in snippet source files
+        test_to_method_key_map: a map from test keys to test data
+                                (filepaths and names) generated by the
+                                language-specific parsers
+    """
+    for method in source_methods:
+        source_root = os.path.dirname(method.source_path)
+
+        keys = []
+        if method.parser == 'direct_invocation':
+            keys = [
+                method.class_name +
+                lib_constants.KEY_SEPARATOR +
+                method.method_name
+            ]
+        elif method.parser == 'webapp2_router':
+            keys = [
+                method.http_methods[0] +
+                lib_constants.KEY_SEPARATOR +
+                method.url
+            ]
+        elif method.parser == 'flask_router':
+            keys = [http_method + lib_constants.KEY_SEPARATOR + method.url
+                    for http_method in method.http_methods]
+
+        new_test_methods = list(method.test_methods)  # deep copy
+        for key in keys:
+            if key in test_to_method_key_map:
+                if test_to_method_key_map[key]:
+                    matching_tests = [
+                        test for test in test_to_method_key_map[key]
+                        if source_root in test[0]
+                    ]
```

Comment:
- Unpacking tuples in generators helps remember what the values inside are. This might be wrong but something like:

`matching_tests = [filepath, name for filepath, name in tests if source_root in filepath]`

Reply:
Good point - done.